src/zenml/artifact_stores/base_artifact_store.py

```diff
@@ -11,11 +25,50 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 #  or implied. See the License for the specific language governing
 #  permissions and limitations under the License.
-from abc import ABC, abstractmethod
 
-from zenml.enums import ArtifactStoreFlavor, StackComponentType
+import textwrap
+from abc import ABC
+from typing import (
+    Any,
+    Callable,
+    ClassVar,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Set,
+    Tuple,
+    Union,
+)
+
+from pydantic import root_validator
+from tfx.dsl.io.fileio import NotFoundError
+
+from zenml.enums import StackComponentType
+from zenml.exceptions import ZenMLBaseException
 from zenml.stack import StackComponent
 
+PathType = Union[bytes, str]
+
+
+class ArtifactStoreInterfaceError(ZenMLBaseException):
```

Comment:
I guess this could be moved to all the other exceptions?

Reply:
Great suggestion. Moved the exception.